src/main/scala/mesosphere/marathon/state/Volume.scala

```diff
@@ -17,68 +20,81 @@ object Volume {
     containerPath: String,
     hostPath: Option[String],
     mode: Mesos.Volume.Mode,
-    persistent: Option[PersistentVolumeInfo]): Volume =
+    persistent: Option[PersistentVolumeInfo],
+    external: Option[ExternalVolumeInfo]): Volume = {
+
     persistent match {
       case Some(persistentVolumeInfo) =>
+        if (hostPath.isDefined) throw new IllegalArgumentException("hostPath may not be set with persistent")
```

Comment:
json formats calls this. maybe there's a way to implement a "Reads" (or whatever that arcane library uses) to validate this instead of doing it this way